src/aliceVision/calibration/checkerDetector.cpp

```diff
@@ -16,69 +22,74 @@ template <typename Type, int Size>
 using RowVector = Matrix<Type, 1, Size>;
 } // namespace Eigen
 
-namespace aliceVision{
-namespace calibration{
-
-
+namespace aliceVision {
+namespace calibration {
 
-bool CheckerDetector::process(const image::Image<image::RGBColor> & source)
+bool CheckerDetector::process(const image::Image<image::RGBColor>& source, bool useNestedGrids, bool debug)
 {
     image::Image<float> grayscale;
     image::ConvertPixelType(source, &grayscale);
 
+    const Vec2 center(grayscale.Width() / 2, grayscale.Height() / 2);
+
     const double scales[] = {1.0, 0.75, 0.5, 0.25};
 
-    std::vector<Vec2> allCorners;
+    std::vector<IntermediateCorner> allCorners;
     for (double scale : scales)
     {
+        ALICEVISION_LOG_INFO("[CheckerDetector] extracting corners at scale " << scale);
         std::vector<Vec2> corners;
         if (!processLevel(corners, grayscale, scale))
         {
+            ALICEVISION_LOG_DEBUG("[CheckerDetector] detection failed");
             return false;
         }
 
-        //Merge with previous level corners
+        ALICEVISION_LOG_DEBUG("[CheckerDetector] detected " << corners.size() << " corners");
+
+        // Merge with previous level corners
         for (Vec2 c : corners)
         {
-            double distmin = 5.0;
+            bool keep = true;
 
-            for (Vec2 oc : allCorners)
+            for (const IntermediateCorner & oc : allCorners)
             {
-                double dist = (oc - c).norm();
+                double dist = (oc.center - c).norm();
                 
-                if (dist < distmin)
+                if (dist < 5.0)
```

Comment:
If this is some arbitrary threshold it would be better to use a const variable declared at the very beginning in case one day it has to be exposed in the API
```c++
const double minCornerDist{5.};
...
    if (dist < minCornerDist)
```